### PR TITLE
add poetry files, and fix numpy error and no data error.

### PR DESCRIPTION
**Pull request by MasahiroOgawa**
# Objective
Fix [the issue](https://github.com/nerfstudio-project/gsplat/issues/239); fix numpy version and add download data script.
The user will not bother with the same issue with me.

### CODE REVIEW

**Comment by Atticuszz** [comment]
nice pyproject.toml.😝

**Comment by maturk** [comment]
Hi, I have a few thoughts. Maybe you can move the download dataset script into gsplat/examples/datasets subfolder. Also, I would prefer not to have a poetry.lock, this is too restrictive and just adds another thing that should be carefully checked and also maintained for compatibility for various different machines. Also, gsplat can be used as a dependency which does not require the installation of the various requirements in the examples/ directory. I think we should not have both a setup.py and a pyproject. The project should support various python versions (currently the lock is set at ^3.11 which is 3.11>=) and cuda/torch versions (currently locked at ^2.3.1+cu121 which requires quite a new torch version >=2.3.1 and toolkit respectively). Do you think just updating the requirements.txt with the correct numpy verison suffices for correct installation of the examples?

**Comment by MasahiroOgawa** [comment]
Got it! I will remove poetry files and move the download script later.

**Comment by maturk** [comment]
I removed the pyproject and added a generic download script that currently just downloads mipnerf360, but can be expanded for other benchmark datasets if needed.

File: examples/datasets/download_dataset.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import subprocess
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Literal
+
+import tyro
+
+# dataset names
+dataset_names = Literal[
+    "mipnerf",
```
Comment (liruilong940607) [inline examples/datasets/download_dataset.py:13]: Would "mipnerf360" make more sense to call it?

**Review by liruilong940607 — APPROVED** [review]
LGTM!

**Comment by MasahiroOgawa** [comment]
Thanks! I read the code. Good download script!

**Pull request merged**